src/main/java/com/gotocompany/firehose/sink/grpc/client/GrpcClient.java

```diff
@@ -41,39 +45,47 @@ public GrpcClient(FirehoseInstrumentation firehoseInstrumentation, GrpcSinkConfi
         this.managedChannel = managedChannel;
     }
 
-    public DynamicMessage execute(byte[] logMessage, Headers headers) {
-
+    public void initialize() {
         MethodDescriptor.Marshaller<byte[]> marshaller = getMarshaller();
-        DynamicMessage dynamicMessage;
-
-        try {
+        this.methodDescriptor = MethodDescriptor.newBuilder(marshaller, marshaller)
```

Comment:
I think we can remove this method and put this codeblock in constructor.

Reply:
```
    public GrpcClient(FirehoseInstrumentation firehoseInstrumentation, GrpcSinkConfig grpcSinkConfig, ManagedChannel managedChannel, StencilClient stencilClient) {
        this.firehoseInstrumentation = firehoseInstrumentation;
        this.grpcSinkConfig = grpcSinkConfig;
        this.stencilClient = stencilClient;
        this.managedChannel = managedChannel;
        MethodDescriptor.Marshaller<byte[]> marshaller = getMarshaller();
        this.methodDescriptor = MethodDescriptor.newBuilder(marshaller, marshaller)
                .setType(MethodDescriptor.MethodType.UNARY)
                .setFullMethodName(grpcSinkConfig.getSinkGrpcMethodUrl())
                .build();
    }
```
You can make AllArgsContructor for testing.

Reply:
Agree